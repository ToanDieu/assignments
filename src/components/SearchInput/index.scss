.search-input {
  margin-bottom: 30px;
  padding-right: 0;
  &__content {
    position: relative;
    .content {
      &__icon {
        position: absolute;
        line-height: 1.5;
        height: 100%;
        padding: 10px;
      }
      &__input {
        padding: 0 20px 0px 40px;
      }
    }
  }
}
