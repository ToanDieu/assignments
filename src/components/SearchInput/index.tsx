import React, { useState } from "react";
import { Form, Container, Row, Col } from "react-bootstrap";

import IconSearch from "../../assets/icons/icon-search.svg";

import "./index.scss";

type Props = {
  onKeyPress: Function;
  placeholder?: string;
};

const SearchInput = ({
  onKeyPress,
  placeholder = "Enter city name",
}: Props) => {
  const [val, setVal] = useState("");
  const onKeyPressHandler = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter") {
      onKeyPress(val);
    }
  };

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setVal(e.target.value);
  };

  return (
    <Container className={"search-input"}>
      <Row className={"search-input__content content"}>
        <Col sm={8} md={8} lg={8}></Col>
        <Col sm={4} md={4} lg={4}>
          <div>
            <img
              className={"content__icon"}
              alt="icon serach"
              src={IconSearch}
            />
            <Form.Control
              onChange={onChange}
              onKeyPress={onKeyPressHandler}
              className={"content__input"}
              value={val}
              placeholder={placeholder}
            />
          </div>
        </Col>
      </Row>
    </Container>
  );
};

export default React.memo(SearchInput);
