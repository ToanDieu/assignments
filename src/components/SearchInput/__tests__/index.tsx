import { render, fireEvent, cleanup } from "@testing-library/react";

import SearchInput from "../index";

afterEach(cleanup);

describe("<SearchInput/>", () => {
  test("onKeyPress Function should fired", () => {
    const handler = jest.fn();
    const { container } = render(<SearchInput onKeyPress={handler} />);
    const input =
      container.querySelector(".content__input") ||
      document.createElement("input");
    fireEvent.keyPress(input, {
      key: "Enter",
      code: 13,
      charCode: 13,
    });
    expect(handler.mock.calls.length).toBe(1);
  });
  test("SearchInput should render and match snapshot", () => {
    const { container } = render(<SearchInput onKeyPress={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
