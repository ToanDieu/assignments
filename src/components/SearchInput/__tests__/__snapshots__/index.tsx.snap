// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput/> SearchInput should render and match snapshot 1`] = `
<div
  class="search-input container"
>
  <div
    class="search-input__content content row"
  >
    <div
      class="col-lg-8 col-md-8 col-sm-8"
    />
    <div
      class="col-lg-4 col-md-4 col-sm-4"
    >
      <div>
        <img
          alt="icon serach"
          class="content__icon"
          src="icon-search.svg"
        />
        <input
          class="content__input form-control"
          placeholder="Enter city name"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
